client/src/portal/api/mod.rs

```diff
@@ -259,6 +259,22 @@ impl Keyring {
         )
     }
 
+    pub(crate) fn verify_secret(&self, secret: &Secret) -> bool {
+        let key = self.derive_key(secret);
+
+        // if this is a new keyring or a keyring with zero items
+        if self.items.is_empty() {
+            return true;
+        }
+
+        // attempting to decrypt the first item
+        if self.items[0].clone().decrypt(&key).is_ok() {
```

Comment:
Regarding style, 

`if let Some(value) = self.items.get(0)` is simpler than asking if the vec is empty and then calling `[0]`.

Reply:
Now, I'm decrypting all the items. Does it still make sense to follow this approach?